crates/bevy_audio/src/sinks.rs

```diff
@@ -15,6 +15,11 @@ pub trait AudioSinkPlayback {
     ///
     /// The value `1.0` is the "normal" volume (unfiltered input). Any value other than `1.0`
     /// will multiply each sample by this value.
+    ///
+    /// # Note on Audio Volume
+    /// Our perception of audio volume is based on a logarithmic scale. As such, setting the volume to 0.01
+    /// doesn't mean it's 100 times quieter. Instead, it's exactly two times quieter than the default.
+    /// Mathematically, if 10^x = 0.01, then x = -2, representing a factor of two.
```

Comment:
```suggestion
    /// An increase of 10 decibels (dB) roughly corresponds to the perceived volume doubling in intensity.
    /// As this function scales not the volume but the amplitude, a conversion might be necessary.
    /// For example, to halve the perceived volume you need to decrease the volume by 10 dB.
    /// This corresponds to 20log(x) = -10dB, solving x = 10^(-10/20) = 0.316.
    /// Multiply the current volume by 0.316 to halve the perceived volume.
```
Technically `set_volume` is misleading as it scales the amplitude, not the volume. Usually audio volume is in dB by definition.

IMO it should be called something like `set_amplitude` and then we can have function like `set_volume_percentage(f32)` where we do the conversion (which will be subjective). But for this PR let's just make sure the comment is correct and helpful. 

@harudagondi have you tried setting the volume to 0.01? What we're saying is that setting it to 0.01 makes the perceived volume wayyy less than two times quieter.

Reply:
Hmm that's true actually. Let's just merge this PR and then add a deprecate warning on `set_volume`, or possibly just have the enum Volume thing like i suggested.